src/Type/Php/CurlGetinfoFunctionDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,202 @@
+			'CURLINFO_HEADER_SIZE' => new FloatType(),
+			'CURLINFO_HEADER_OUT' => TypeCombinator::union(new StringType(), new ConstantBooleanType(false)),
+			'CURLINFO_REQUEST_SIZE' => new FloatType(),
+			'CURLINFO_SSL_VERIFYRESULT' => new IntegerType(),
+			'CURLINFO_CONTENT_LENGTH_DOWNLOAD' => new FloatType(),
+			'CURLINFO_CONTENT_LENGTH_UPLOAD' => new FloatType(),
+			'CURLINFO_CONTENT_TYPE' => new StringType(),
+			'CURLINFO_PRIVATE' => TypeCombinator::union(new StringType(), new ConstantBooleanType(false)),
+			'CURLINFO_RESPONSE_CODE' => new IntegerType(),
+			'CURLINFO_HTTP_CONNECTCODE' => new IntegerType(),
+			'CURLINFO_HTTPAUTH_AVAIL' => new IntegerType(),
+			'CURLINFO_PROXYAUTH_AVAIL' => new IntegerType(),
+			'CURLINFO_OS_ERRNO' => new IntegerType(),
+			'CURLINFO_NUM_CONNECTS' => new IntegerType(),
+			'CURLINFO_SSL_ENGINES' => new ArrayType(new IntegerType(), new StringType()),
+			'CURLINFO_COOKIELIST' => new ArrayType(new IntegerType(), new StringType()),
+			'CURLINFO_FTP_ENTRY_PATH' => TypeCombinator::union(new StringType(), new ConstantBooleanType(false)),
+			'CURLINFO_APPCONNECT_TIME' => new FloatType(),
+			'CURLINFO_CERTINFO' => new ArrayType(new IntegerType(), new ArrayType(new IntegerType(), new StringType())),
+			'CURLINFO_CONDITION_UNMET' => new IntegerType(),
+			'CURLINFO_RTSP_CLIENT_CSEQ' => new IntegerType(),
+			'CURLINFO_RTSP_CSEQ_RECV' => new IntegerType(),
+			'CURLINFO_RTSP_SERVER_CSEQ' => new IntegerType(),
+			'CURLINFO_RTSP_SESSION_ID' => new IntegerType(),
+			'CURLINFO_HTTP_VERSION' => new IntegerType(),
+			'CURLINFO_PROTOCOL' => new StringType(),
+			'CURLINFO_PROXY_SSL_VERIFYRESULT' => new IntegerType(),
+			'CURLINFO_SCHEME' => new StringType(),
+			'CURLINFO_CONTENT_LENGTH_DOWNLOAD_T' => new FloatType(),
+			'CURLINFO_CONTENT_LENGTH_UPLOAD_T' => new FloatType(),
+			'CURLINFO_SIZE_DOWNLOAD_T' => new FloatType(),
+			'CURLINFO_SIZE_UPLOAD_T' => new FloatType(),
+			'CURLINFO_SPEED_DOWNLOAD_T' => new FloatType(),
+			'CURLINFO_SPEED_UPLOAD_T' => new FloatType(),
+			'CURLINFO_APPCONNECT_TIME_T' => new FloatType(),
+			'CURLINFO_CONNECT_TIME_T' => new FloatType(),
+			'CURLINFO_FILETIME_T' => new FloatType(),
+			'CURLINFO_NAMELOOKUP_TIME_T' => new FloatType(),
+			'CURLINFO_PRETRANSFER_TIME_T' => new FloatType(),
+			'CURLINFO_REDIRECT_TIME_T' => new FloatType(),
+			'CURLINFO_STARTTRANSFER_TIME_T' => new FloatType(),
+			'CURLINFO_TOTAL_TIME_T' => new FloatType(),
+		];
+
+		foreach ($componentTypesPairedConstants as $constantName => $type) {
+			$constantNameNode = new Name($constantName);
+			if ($this->reflectionProvider->hasConstant($constantNameNode, $scope) === false) {
+				continue;
+			}
+
+			$valueType = $this->reflectionProvider->getConstant($constantNameNode, $scope)->getValueType();
+			if ($valueType instanceof ConstantIntegerType === false) {
+				throw new ShouldNotHappenException();
+			}
+
+			if ($valueType->getValue() === $componentType->getValue()) {
```

Comment:
I guess this should be

```suggestion
			if ($componentType->getValue()->isSuperTypeOf($valueType)->yes()) {
```

or maybe the other way arround